src/main/java/networkbook/ui/PersonCard.java

```diff
@@ -44,40 +45,71 @@ public class PersonCard extends UiPart<Region> {
     @FXML
     private Label phones;
     @FXML
-    private Label links;
+    private Label linksHeader;
+    @FXML
+    private FlowPane links;
     @FXML
     private Label graduation;
     @FXML
     private Label courses;
     @FXML
     private Label specialisations;
     @FXML
-    private Label emails;
+    private Label emailsHeader;
+    @FXML
+    private FlowPane emails;
     @FXML
     private FlowPane tags;
     @FXML
     private Label priority;
 
     /**
-     * Creates a {@code PersonCode} with the given {@code Person} and index to display.
+     * Creates a {@code PersonCard} with the given {@code Person} and index to display.
      */
     public PersonCard(Person person, int displayedIndex) {
         super(FXML);
         requireNonNull(person);
         this.person = person;
+
+        // Name and ID
         id.setText(displayedIndex + ". ");
         name.setText(person.getName().fullName);
+
+        // Phone numbers
         phones.setText(PHONES_HEADER + person.getPhones().toString());
-        emails.setText(EMAILS_HEADER + person.getEmails().toString());
-        links.setText(LINKS_HEADER + person.getLinks().toString());
+
+        // Email addresses
+        emailsHeader.setText(EMAILS_HEADER);
+        // TODO: implement actual link opening
+        person.getEmails().stream()
+                .forEach(email -> emails.getChildren().add(new FieldHyperlink(email.getValue(), () -> {
+                    System.out.println("open email: " + email.getValue());
```

Comment:
I assume that this will be replaced by the actual code to open link/email. Ideally should use logger, KIV.

Reply:
> I assume that this will be replaced by the actual code to open link/email. Ideally should use logger, KIV.

Yup, the println is just a placeholder and should be replaced by the actual link opening code.

---

Thank you for the feedback! I am aware that the changes are rather minimal at the moment, and will definitely improve on the UI further in my next PR. This PR is indeed mainly to accommodate opening emails and links so that the rest of the UI development (which may take longer) does not hinder the link opening feature's development.

I think that your suggestions are good and will implement them (as well as the other UI features in the Figma) in the next PR. For now, I will make the change in javafx version that you have mentioned and add some documentation for the UI in the DG.